python/cuml/utils/numba_utils.py

```diff
@@ -183,3 +184,25 @@ def stride_from_order(shape, order, itemsize):
     if order == 'C':
         stride.reverse()
     return tuple(stride)
+
+
+# CuPy < 7.0 and numba >= 0.46 mitigation
+# Remove after making CuPy >= 7 required!
+
+
+class PatchedNumbaDeviceArray(object):
+    def __init__(self, numba_array):
+        self.parent = numba_array
+
+    def __getattr__(self, name):
```

Comment:
I don't think getattr is enough to be a full proxy. It won't support subscripting and it'll behave differently when checked for "trueness". Being an almost-proxy may be more confusing than just providing an explicit function...

Are we literally only using this for cp.unique and similar functions right now? While we want for the fix and upgrade to cupy 7.x, we could just provide cupy-based unique as a wrapper without exposing this proxy object.

maybe you could just provide "get_cuda_array"

What about ditching the proxy and just making a "get_cuda_array_interface" method